ui/src/core/xibo-cms.js

```diff
@@ -3079,6 +3087,103 @@ function createColorPicker(element, options) {
   }, options));
 }
 
+
+/**
+ * Create custom date range filter
+ * @param {object} element jquery object or CSS selector
+ */
+function updateDateRangeFilter(element) {
+  const selected = $(element).find('option:selected').val();
+  const $form = $(element).closest('form');
+
+  // This would allow us to target this specific element in case there are multiple range filters used
+  const selectFilterId = $(element).attr('id');
+
+  let fromDt = moment().startOf('day');
+  let toDt = moment().endOf('day');
+
+  // Remove the hidden attribute
+  $(`.rangeFilterInput_${selectFilterId}`).removeClass('hidden');
+
+  // Hide/Show From/To Date
+  if (selected === '' || selected === undefined) {
+    $form.find(`.rangeFilterInput_${selectFilterId}`).show();
+
+    updateDateRangeSelection($form, selectFilterId);
+  } else {
+    $form.find(`.rangeFilterInput_${selectFilterId}`).hide();
+
+    switch (selected) {
+      case 'yesterday':
+        fromDt = moment().startOf('day').subtract(1, 'days');
+        toDt = moment().endOf('day').subtract(1, 'days');
+        break;
+      case 'thisweek':
+        fromDt = moment().startOf('week');
+        toDt = moment().endOf('week');
+        break;
+      case 'thismonth':
+        fromDt = moment().startOf('month');
+        toDt = moment().endOf('month');
+        break;
+      case 'thisyear':
+        fromDt = moment().startOf('year');
+        toDt = moment().endOf('year');
+        break;
+      case 'lastweek':
+        fromDt = moment().startOf('week').subtract(1, 'weeks');
+        toDt = moment().endOf('week').subtract(1, 'weeks');
+        break;
+      case 'lastmonth':
+        fromDt = moment().startOf('month').subtract(1, 'months');
+        toDt = moment().endOf('month').subtract(1, 'months');
+        break;
+      case 'lastyear':
+        fromDt = moment().startOf('year').subtract(1, 'years');
+        toDt = moment().endOf('year').subtract(1, 'years');
+        break;
+    }
+  }
+
+  formatInputFields($form, selectFilterId, fromDt, toDt);
+}
+
+/**
+ * Calculates from/to date range selection
+ * @param {object} $form jquery object or CSS selector
+ * @param {string} selectFilterId filter ID selector
+ */
+function updateDateRangeSelection($form, selectFilterId) {
+  let fromDt = moment().startOf('day');
+  let toDt = moment().endOf('day');
+
+  $(`.rangeFilterInput_${selectFilterId}`).on('change', function() {
+    fromDt = moment($(`#fromDt_${selectFilterId}`).val()).startOf('day').format();
```

Comment:
Same thing as the function above (https://eslint.org/docs/latest/rules/max-len).